front/lib/amplitude/node/index.ts

```diff
@@ -185,6 +186,38 @@ export function trackDataSourceCreated(
   );
 }
 
+export function trackDataSourceUpdated(
+  auth: Authenticator,
+  {
+    dataSource,
+  }: {
+    dataSource: DataSourceType;
+  }
+) {
+  const userId = auth.user()?.id;
+  const workspace = auth.workspace();
+  if (!workspace || !userId) {
+    return;
+  }
+  const amplitude = getBackendClient();
+  const event = new DataSourceUpdated({
+    dataSourceName: dataSource.name,
+    dataSourceProvider: dataSource.connectorProvider || "",
+    workspaceName: workspace.name,
+    workspaceId: workspace.sId,
+    assistantDefaultSelected: dataSource.assistantDefaultSelected,
+  });
+
+  amplitude.track(
+    `user-${userId}`,
+    { ...event, groups: { [GROUP_TYPE]: workspace.sId } },
+    {
+      time: Date.now(),
+      insert_id: `data_source_updated_${dataSource.id}`,
```

Comment:
`insert_id` is a deduplication key to make sure Amplitude won't process an event twice if there is an error over the connection while sending it and it's being re-sent by the client, or if we replay it because we had a crash of some sort.

The problem here is that each update of the data source would result in the same dedup key since it's datasource.id does not change.
I think the deduplication key in this case should be:
`data_source_updated_${dataSource.id}_${dataSource.updatedAt}`

Reply:
Thanks I added the current timestamp 🙏🏻